sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserInterface.scala

```diff
@@ -70,4 +71,10 @@ trait ParserInterface {
    */
   @throws[ParseException]("Text cannot be parsed to a DataType")
   def parseDataType(sqlText: String): DataType
+
+  /**
+   * Parse a string to a [[CalendarInterval]].
+   */
+  @throws[ParseException]("Text cannot be parsed to an interval")
+  def parseInterval(sqlText: String): CalendarInterval
```

Comment:
@cloud-fan . This looks like a breaking change to the existing `Parser` Spark Extension. If then, this need a new JIRA. Can we have this change in a separate JIRA PR because this is an open developer API?

Reply:
I didn't realize that it's a developer API. I think it's fine to update the developer APIs, and we already did it when adding `parseMultipartIdentifier`.

Since this PR doesn't need to access `ParserInterface` directly, let me avoid changing `ParserInterface`

Reply:
Thanks! Of course, we can change it~ What we need is just having two separate PRs to distinguish them clearly.